### PR TITLE
Add new constructor to cosmtap2DROS

### PR DESCRIPTION
**Pull request by tonynajjar**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | [(3213)](https://github.com/ros-planning/navigation2/issues/3213) |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Gazebo |

---

## Description of contribution in a few bullet points

* Added a new constructor in cosmtap2DROS to allow setting namespace and other options from launch file

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by tonynajjar** [comment]
@SteveMacenski I'm not entirely sure I understood what you meant in the [comment](https://github.com/ros-planning/navigation2/issues/3213#issuecomment-1255323699). Please let me know if I'm far off

File: nav2_costmap_2d/src/costmap_2d_ros.cpp [hunk 1]
```diff
@@ -84,6 +98,15 @@ Costmap2DROS::Costmap2DROS(
     "nav2_costmap_2d::StaticLayer",
     "nav2_costmap_2d::ObstacleLayer",
     "nav2_costmap_2d::InflationLayer"}
+{
+  declare_parameter(
```
Comment (SteveMacenski) [inline nav2_costmap_2d/src/costmap_2d_ros.cpp:102]: Why is `plugins` singled out? That appears the same either way.

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: nav2_costmap_2d/src/costmap_2d_ros.cpp (hunk 1 above)
Comment (tonynajjar) [reply nav2_costmap_2d/src/costmap_2d_ros.cpp:102]: Just didn't want to pass it to init() but I can

Comment (tonynajjar) [reply nav2_costmap_2d/src/costmap_2d_ros.cpp:101]: Oops, you're right, fixed

**Review by tonynajjar — COMMENTED** [review]
2 questions:

Can we now remove the (3-years old) TODO from @orduno? I'm not sure what is meant with `Pass a sub-node instead of creating a new node` 

Should this constructor be the default for the standalone node?

**Comment by SteveMacenski** [comment]
Should this new constructor be used in the node main file?

**Comment by tonynajjar** [comment]
> Should this new constructor be used in the node main file?

Jinx :grin: I would think so yes, it can be confusing when the name and namespaces are not being set from the launch file

**Comment by SteveMacenski** [comment]
Just waiting on the main updates and I'll merge. Please prepare a navigation.ros.org PR to detail the change in the migration guide

**Comment by tonynajjar** [comment]
> Just waiting on the main updates and I'll merge

Ah ok was waiting for confirmation. main updated

> Please prepare a navigation.ros.org PR to detail the change in the migration guide

You mean to detail the changes needed in the launch files for people using the standalone node?

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Detail that the costmap standalone node has changed, and yes, if any changes are needed to get the previous behavior, what those are

**Pull request merged**